azavea/cac-tripplanner
# Switch carousel library

## Overview

Replace slick-carousel with tiny-slider.

Updates to the JavaScript build broke use of the carousel library. The exact cause was unclear, and the `slick-carousel` library looks to be effectively unsupported, so here it is replaced with another carousel library, `tiny-slider`, which does not have external dependencies and has recent updates.

The new carousel should look and function just as the old one did.


### Demo

![image](https://user-images.githubusercontent.com/960264/58135974-651dc100-7bfa-11e9-8771-ce629d38cdad.png)

Mobile itineraries:
![image](https://user-images.githubusercontent.com/960264/58136125-f725c980-7bfa-11e9-9999-2bcbd70d24ae.png)

Itinerary cards resize height on swipe (using `autoHeight`):
![image](https://user-images.githubusercontent.com/960264/58136150-13296b00-7bfb-11e9-8c66-fb84976a47e0.png)


### Notes

If the map page gets loaded at a larger width than the mobile breakpoint, then is resized to be narrower than the mobile breakpoint, the dots don't load under the itinerary carousel, although the carousel otherwise renders and functions as expected. This should be fine, as it is unlikely in normal use that a user would resize their desktop browser to be mobile-sized, and the functionality is still present.


## Testing Instructions

 * Add multiple images to at least one destination 
 * Add at least one event with multiple images
 * Carousel should render and autoplay as expected on the place/event detail pages
 * The single image for places/events without multiple images should still display as expected
 * Directions on mobile (<480px width) should display itineraries in carousel as expected

## Checklist


Fixes #1102
